lua/nvim-tree/renderer/init.lua

```diff
@@ -63,34 +66,39 @@ end
 local get_file_icon = function()
   return ""
 end
-if icon_state.show_file_icon then
-  if icon_state.has_devicons then
-    local web_devicons = require "nvim-web-devicons"
 
-    get_file_icon = function(fname, extension, line, depth)
-      local icon, hl_group = web_devicons.get_icon(fname, extension)
+local get_file_icon_default = function(_, _, line, depth)
+  local hl_group = "NvimTreeFileIcon"
+  local icon = icon_state.icons.default
+  if #icon > 0 then
+    table.insert(hl, { hl_group, line, depth, depth + #icon + 1 })
+  end
+  return #icon > 0 and icon .. icon_padding or ""
+end
 
-      if icon and hl_group ~= "DevIconDefault" then
-        if hl_group then
-          table.insert(hl, { hl_group, line, depth, depth + #icon + 1 })
-        end
-        return icon .. icon_padding
-      elseif string.match(extension, "%.(.*)") then
-        -- If there are more extensions to the file, try to grab the icon for them recursively
-        return get_file_icon(fname, string.match(extension, "%.(.*)"), line, depth)
-      else
-        return #icon_state.icons.default > 0 and icon_state.icons.default .. icon_padding or ""
```

Comment:
This has been replaced by `get_file_icon_default`, so that we get the hl group colour on files that do not have a web dev icon.